tests/chebyshev_test.py

```diff
@@ -0,0 +1,795 @@
+import numpy as np
+from numpy.polynomial import chebyshev as ncs
+import numpy.polynomial.polyutils as npu
+from numpy.polynomial.polyutils import RankWarning
+
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from jax import numpy as jnp
+from jax._src import test_util as jtu
+import jax.numpy.polynomials as pu
+from jax.numpy.polynomials import chebyshev as cs
+
+from jax.config import config
+config.parse_flags_with_absl()
+
+if config.jax_enable_x64:
+  all_dtypes = ["int64", "float64", "complex128"]
+else:
+  all_dtypes = ["int32", "float32", "complex64"]
+
+class TestChebyshev(jtu.JaxTestCase):
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_dtype={}_mul{}".format(dtype, mul),
+    "dtype": dtype, "mul": mul}
+    for dtype in all_dtypes
+    for mul in [0,1]))
+  def testChebline(self, dtype, mul):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      off, scl = rng(2, dtype)
+      scl *= mul
+      return (off, scl)
+
+    jnp_fn = lambda off, scl: cs.chebline(off, scl)
+    np_fn = lambda off, scl: ncs.chebline(off, scl)
+    self._CheckAgainstNumpy(np_fn, jnp_fn, args_maker, check_dtypes=False,
+                            tol=1e-6)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": ("_dtype={}_leading1={}_trailing1={}"
+    "_dtype={}_leading2={}_trailing2={}").format(
+      jtu.format_shape_dtype_string((length1+leading1+trailing1,), dtype),
+      leading1, trailing1,
+      jtu.format_shape_dtype_string((length2+leading2+trailing2,), dtype),
+      leading2, trailing2),
+      "dtype": dtype, "length1": length1, "leading1": leading1, "trailing1": trailing1,
+      "length2": length2, "leading2": leading2, "trailing2": trailing2}
+      for dtype in all_dtypes
+      for length1 in [1, 3, 9, 10, 17]
+      for length2 in [1, 3, 9, 10, 17]
+      for leading1 in [0, 1, 2, 7]
+      for leading2 in [0, 1, 2, 7]
+      for trailing1 in [0, 1, 2, 7]
+      for trailing2 in [0, 1, 2, 7]))
```

Comment:
This test generates a HUGE number of test cases (3 * 5 * 5 * 4 * 4 * 4 * 4 = 19200). The most we ever check in CI is 50 (via `jtu.cases_from_list`), but all 19200 pairs will be generated when the test file is loaded. That tends to lead to needlessly long test load times. Consider reducing the number of test cases you generate. Same comment applies below.